lib/easy.c

```diff
@@ -133,6 +182,9 @@ static char *leakpointer;
  */
 static CURLcode global_init(long flags, bool memoryfuncs)
 {
+  if ((flags & CURL_GLOBAL_NEED_THREADSAFE) && !GLOBAL_INIT_IS_THREADSAFE)
+    return CURLE_NOT_THREADSAFE;
```

Comment:
For the documentation it should be noted that passing in `CURL_GLOBAL_NEED_THREADSAFE` when the application is using an older libcurl (from before this support is added) may still have libcurl use an unsafe global_init.

Reply:
I have added a new flag in curl_version_info_data.features instead.